File: c/problem3.c
#include <stdio.h>


/*	https://projecteuler.net/problem=3	*/
int main()
{
	unsigned long M = 600851475143;
	unsigned int n = 2;

	do
	{
		if ( M % n != 0 )
		{
			n++;
		}
		else
		{
			M /= n;
		}

	} while ( n * n <= M );

	// result
	printf("%lu", M);

	return 0;
}

File: matlab/problem3.m
clear; close all; clc
%% https://projecteuler.net/problem=3

N = 600851475143;
n = 2;

% measure execution speed
tic
while n*n <= N
    if mod(N, n) ~= 0
        n = n + 1;
    else
        N = floor(N / n);
    end
end
toc
% Elapsed time is 0.001267 seconds.

% result
disp(N)